jexl-eval/src/lib.rs

```diff
@@ -67,103 +71,214 @@ impl Truthy for Value {
-                        }
-                    })
+            Expression::BinaryOperation {
+                left,
+                right,
+                operation,
+            } => {
+                let left = self.eval_ast(*left, context)?;
+                let right = self.eval_ast(*right, context)?;
+                match (operation, left, right) {
+                    (OpCode::And, a, b) => Ok(if a.is_truthy() { b } else { a }),
+                    (OpCode::Or, a, b) => Ok(if a.is_truthy() { a } else { b }),
+
+                    (op, Value::Number(a), Value::Number(b)) => {
+                        let left = a.as_f64().unwrap();
+                        let right = b.as_f64().unwrap();
+                        Ok(match op {
+                            OpCode::Add => value!(left + right),
+                            OpCode::Subtract => value!(left - right),
+                            OpCode::Multiply => value!(left * right),
+                            OpCode::Divide => value!(left / right),
+                            OpCode::FloorDivide => value!((left / right).floor()),
+                            OpCode::Modulus => value!(left % right),
+                            OpCode::Exponent => value!(left.powf(right)),
+                            OpCode::Less => value!(left < right),
+                            OpCode::Greater => value!(left > right),
+                            OpCode::LessEqual => value!(left <= right),
+                            OpCode::GreaterEqual => value!(left >= right),
+                            OpCode::Equal => value!((left - right).abs() < EPSILON),
+                            OpCode::NotEqual => value!((left - right).abs() > EPSILON),
+                            OpCode::In => value!(false),
+                            OpCode::And | OpCode::Or => {
+                                unreachable!("Covered by previous case in parent match")
+                            }
+                        })
+                    }
+
+                    (OpCode::Add, Value::String(a), Value::String(b)) => {
+                        Ok(value!(format!("{}{}", a, b)))
+                    }
+                    (OpCode::In, Value::String(a), Value::String(b)) => Ok(value!(b.contains(&a))),
+                    (OpCode::Equal, Value::String(a), Value::String(b)) => Ok(value!(a == b)),
+                    (operation, left, right) => Err(EvaluationError::InvalidBinaryOp {
+                        operation,
+                        left,
+                        right,
+                    }),
                 }
-
-                (OpCode::Add, Value::String(a), Value::String(b)) => {
-                    Ok(value!(format!("{}{}", a, b)))
+            }
+            Expression::Transform {
+                name,
+                subject,
+                args,
+            } => {
+                let subject = self.eval_ast(*subject, context)?;
+                let mut args_arr = Vec::new();
```

Comment:
This mimics how the js stuff is implemented. We evaluate the subject, then add it as the first argument